pkg/envoy/proxy.go

```diff
@@ -70,6 +70,10 @@ func NewProxy(cfg ProxyConfig) Proxy {
 		args = append(args, "--component-log-level", cfg.ComponentLogLevel)
 	}
 
+	// Explicitly enable core dumps. This may be desirable more often (by default), but for now we only set it in VM tests.
+	if enableEnvoyCoreDump {
+		args = append(args, "--enable-core-dump")
```

Comment:
Is this off-by-default? Why do we need this flag now?

Reply:
TBH I have no clue. It works on k8s without the flag, and only works with this crazy setup with the flag...

Reply:
let's follow up, maybe file an issue why this isn't the default